packages/protocol/contracts/L1/ProverPool.sol

```diff
@@ -380,7 +380,7 @@ contract ProverPool is EssentialContract, IProverPool {
         // by keep rewardPerGas = 1
         provers[staker.proverId] = Prover(0, 1, 0);
 
-        if (prover.stakedAmount > 0) {
+        if (prover.stakedAmount > 0 || !checkExitTimestamp) {
```

Comment:
emm what will happen if that replaced prover was slashed to 0? (`replacedProver.stakeAmount == 0`)  will his `proverId` still not be reset?